src/test/java/org/broadinstitute/hellbender/utils/MannWhitneyUUnitTest.java

```diff
@@ -1,50 +1,178 @@
+        makeDistribution(distribution20_40, 40, skew, observations/2);
+
+        // shuffle the observations
+        Utils.resetRandomGenerator();
+        Collections.shuffle(distribution20, Utils.getRandomGenerator());
+        Collections.shuffle(distribution30, Utils.getRandomGenerator());
+        Collections.shuffle(distribution20_40, Utils.getRandomGenerator());
+
+        for ( final int numToReduce : Arrays.asList(0, 10, 50, 100) ) {
+            tests.add(new Object[]{distribution20, distribution20, numToReduce, true, observations, "20-20"});
+            tests.add(new Object[]{distribution30, distribution30, numToReduce, true, observations, "30-30"});
+            tests.add(new Object[]{distribution20_40, distribution20_40, numToReduce, true, observations, "20/40-20/40"});
+
+            tests.add(new Object[]{distribution20, distribution30, numToReduce, false, observations, "20-30"});
+            tests.add(new Object[]{distribution30, distribution20, numToReduce, false, observations, "30-20"});
+
+            tests.add(new Object[]{distribution20, distribution20_40, numToReduce, false, observations, "20-20/40"});
+            tests.add(new Object[]{distribution30, distribution20_40, numToReduce, true, observations, "30-20/40"});
         }
-        for ( double dp : d2 ) {
-            mwu.add(dp,MannWhitneyU.USet.SET2);
+
+        return tests.toArray(new Object[][]{});
+    }
+
+    private static void makeDistribution(final List<Double> result, final int target, final int skew, final int numObservations) {
+        final int rangeStart = target - skew;
+        final int rangeEnd = target + skew;
+
+        double current = rangeStart;
+        for ( int i = 0; i < numObservations; i++ ) {
+            result.add(current++);
+            if ( current > rangeEnd )
+                current = rangeStart;
         }
-        final Pair<Double, Double> pair = mwu.runOneSidedTest(MannWhitneyU.USet.SET1);
-        return pair.getLeft();
     }
 
+    @Test(dataProvider = "DistributionData")
+    public void testDistribution(final List<Double> distribution1, final List<Double> distribution2, final int numToReduceIn2, final boolean distributionsShouldBeEqual, final int observations, final String debugString) {
+        final MannWhitneyU mannWhitneyU = new MannWhitneyU();
+
+        final List<Double> dist2 = new ArrayList<>(distribution2);
+        if ( numToReduceIn2 > 0 ) {
+            Double counts = 0.0;
+            Double quals = 0.0;
+
+            for ( int i = 0; i < numToReduceIn2; i++ ) {
+                counts++;
+                quals += dist2.remove(0);
+            }
+
+            final Double qual = quals / counts;
+            for ( int i = 0; i < numToReduceIn2; i++ )
+                dist2.add(qual);
+        }
+
+        final Double result = mannWhitneyU.test(Doubles.toArray(distribution1), Doubles.toArray(dist2), MannWhitneyU.TestType.TWO_SIDED).getP();
```

Comment:
Well, we already do have good test cases for the z-score in the `FIRST_DOMINATES` cases -- this is `testOnesidedZ()`. This additional test case in `testDistribution()` was salvaged from GATK3's `RankSumUnitTest` without any effort to generalize it.

Given the time constraints on this branch, and given that we do have test coverage on the combination of options that's actually used to generate the annotations, I'm inclined to leave `testDistribution()` as-is for now. If you feel strongly that it should be expanded, could you create a separate ticket in this repo for that task? Thanks!